### ISSUE TITLE
Add config to charge a transaction fee

### ISSUE THREAD

**Issue by bumi** [getAlby/lndhub.go#358]
Currently only the lightning fees are charged. As an operator I want to be able to charge a fee per transaction. 
This could be a fixed fee or a percentage. 

This should be the place where the fee transaction entry is created. Currently we set the invoice fee there. 
https://github.com/getAlby/lndhub.go/blob/main/lib/service/invoices.go#L313

the invoice fee is set here: https://github.com/getAlby/lndhub.go/blob/main/lib/service/invoices.go#L239

this could be one function similar to [CalcFeeLimit](https://github.com/getAlby/lndhub.go/blob/79b99c9aa4b7d0c85fc21f71fbf4dae5582c93b1/lib/service/user.go#L109). It's on the Service thus it can be used to set the invoice Fee.

ToDo: check how it is for keysend payments

**Comment by MoritzKa** [comment]
From a business perspective I want to charge different fees 
- for incoming vs. outgoing transactions 
- to a subsegment of users

**Comment by bumi** [comment]
Assigning all the people just as a ping to think about how this option should be implemented.

**Comment by MoritzKa** [comment]
Fee proposal:
- 0%  internal transaction
- 0% + network fees for 0-250k sat volume per month external transaction
- 0.9% or 1 sat for >250k sat volume per month external transaction
(to be discussed)

Fees can dependent on 
- transaction size  
- transaction volume per week/month/etc. 
- recipient (Alby or non-Alby user) 
- sender (Alby or non-Alby user) 
- type of user (e.g. we could offer reduced fees to users who pay a monthly subscription) 

Implementation
1. Step: check payment volume 
2. Step: make fees configurable based on payment volume

**Comment by kiwiidb** [comment]
On a technical level I advocate for a subscription-based model for power users like BTCPayserver, and charge no transaction-based fees. If we go through with the first step of https://github.com/getAlby/alby-deployment/issues/170, then we have a single place to manage users (the rails app). Then we can set transaction limits for these users there, and have a system to manage subscriptions. Based on the subscription payment we can then allow for different transaction volumes easily in LNDhub. We can then have different monthly tiers. And we can add other benefits for paying users (more nostr / lnurl features, access to more apps, etc).

If we have to calculate transaction-based fees in LNDhub then I see these downsides:

- users will complain if they are too high
- complex to implement, putting even more code in the LNDhub codebase

**Comment by raucao** [comment]
@delfme I think what you're asking for is already implemented in the current fee configs (since 1.0):

https://github.com/getAlby/lndhub.go/pull/392/files

(Someone should close this issue, since the tx fee config and feature has been released already.)

**Comment by delfme** [comment]
Thanks @raucao it is not clear to me if Alby api would allow us to specify a transaction fee, which would be handled automatically. The use case is simple: we intermediate the tansfer between two users of our app and take a little transaction fee for the service.

**Comment by raucao** [comment]
Yes, see `SERVICE_FEE`: https://github.com/getAlby/lndhub.go?tab=readme-ov-file#available-configuration

(It seems like the max fee from the PR I linked is still missing from the README doc, so you could add it there, too.)

**Comment by delfme** [comment]
This is the use case: a messaging app where users can also transfer money in-app without having to input addresses for each transaction or handle any complexity. They just tap over the recipient avatar and send money. The fee will be 0.1% capped to $1, which means it is almost zero and user would pay fee for enjoying the seamless experience. We would love to use Alby sdk for the sake of simplicity but it seems it doesn't let us set a "fee" inside client.keysend. What we would need is:
- specifying a transaction/routing fee in client.keysend
- receiving the fee to our developer address 
Is that feasible with Alby? Is there an example or doc?

**Comment by MoritzKa** [comment]
@delfme please contact me at moritz@getalby.com. I'll give you an intro to Alby's tools specifically for your use case.
